crates/sdk-core/src/antithesis.rs

```diff
@@ -0,0 +1,62 @@
+//! Antithesis SDK integration for invariant testing.
+//!
+//! This module provides assertion macros that integrate with the Antithesis
+//! testing platform to detect invariant violations during fuzz testing.
+//!
+//! When the `antithesis_assertions` feature is disabled, assertions compile
+//! to no-ops.
+
+#[cfg(feature = "antithesis_assertions")]
+mod enabled {
+    use std::sync::OnceLock;
+
+    /// Ensure Antithesis is initialized exactly once.
+    pub(crate) fn ensure_init() {
+        static INIT: OnceLock<()> = OnceLock::new();
+        INIT.get_or_init(|| {
+            ::antithesis_sdk::antithesis_init();
+        });
+    }
+
+    /// Assert that a condition is always true during Antithesis fuzz testing.
+    /// Use `false` as the condition to log an invariant violation.
+    macro_rules! assert_always {
+        ($condition:expr, $message:literal, $details:expr) => {{
+            $crate::antithesis::ensure_init();
+            let details: ::serde_json::Value = $details;
+            ::antithesis_sdk::assert_always!($condition, $message, &details);
+        }};
+        ($condition:expr, $message:literal) => {{
+            $crate::antithesis::ensure_init();
+            ::antithesis_sdk::assert_always!($condition, $message);
+        }};
+    }
+
+    pub(crate) use assert_always;
+}
+
+#[cfg(not(feature = "antithesis_assertions"))]
+mod disabled {
+    /// No-op when feature is disabled.
+    #[inline(always)]
+    pub(crate) fn ensure_init() {}
+
+    /// No-op assertion when feature is disabled.
+    macro_rules! assert_always {
+        ($condition:expr, $message:literal, $details:expr) => {{
+            let _ = ($condition, $message);
+            let _ = $details;
+        }};
+        ($condition:expr, $message:literal) => {{
+            let _ = ($condition, $message);
```

Comment:
I think assigning things like this is actually going to cause the compiler not to optimize some stuff away.

Honestly, I'm not sure we want to do this technique at all. It's slightly more wordy, but guarding every call to `assert_always` with the flag at the callsite is safer in terms of actually making sure everything gets compiled out